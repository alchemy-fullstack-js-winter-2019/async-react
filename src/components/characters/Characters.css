ul li {
  list-style: none;
  display: inline-block;
  margin: 2vw;
}

.characters {
  float: left;
  width: 65vw;
}

img {
  width: 200px;
}