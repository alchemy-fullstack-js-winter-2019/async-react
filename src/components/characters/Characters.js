import Character from './Character';
import Results from '../paging/Results';
import { getCharacters } from '../../services/rickAndMortyApi';

export default Results(Character, getCharacters);
