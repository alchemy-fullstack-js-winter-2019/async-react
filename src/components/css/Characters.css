ul {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
}

li {
  list-style-type: none;
  position: relative;
  width: 50%;
}

h2 {
  font-size: 1.1em;
  width: 250px;
}