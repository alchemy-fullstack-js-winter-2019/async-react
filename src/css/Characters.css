.characters {
  font-family: sans-serif;
  ul {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    color: white;
    
    li {
      padding: 20px;
      background: indigo;
      list-style: none;
      border: 5px black;
      width: 200px;
      margin: 10px;
    }
  }
}
