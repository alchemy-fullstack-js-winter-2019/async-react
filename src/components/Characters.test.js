import React from 'React';
import renderer from 'react-test-renderer';
import Characters from './characters/Characters';

describe('Characters', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(<Characters />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
