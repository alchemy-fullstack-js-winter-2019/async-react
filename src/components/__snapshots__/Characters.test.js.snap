// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshot 1`] = `
<ul>
  <li>
    <h2>
      Rick Sanchez
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Morty Smith
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Summer Smith
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Female
    </p>
  </li>
  <li>
    <h2>
      Beth Smith
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Female
    </p>
  </li>
  <li>
    <h2>
      Jerry Smith
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Abadango Cluster Princess
    </h2>
    <p>
      Alive
    </p>
    <p>
      Alien
    </p>
    <p>
      Female
    </p>
  </li>
  <li>
    <h2>
      Abradolf Lincler
    </h2>
    <p>
      unknown
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Adjudicator Rick
    </h2>
    <p>
      Dead
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Agency Director
    </h2>
    <p>
      Dead
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Alan Rails
    </h2>
    <p>
      Dead
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Albert Einstein
    </h2>
    <p>
      Dead
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Alexander
    </h2>
    <p>
      Dead
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Alien Googah
    </h2>
    <p>
      unknown
    </p>
    <p>
      Alien
    </p>
    <p>
      unknown
    </p>
  </li>
  <li>
    <h2>
      Alien Morty
    </h2>
    <p>
      unknown
    </p>
    <p>
      Alien
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Alien Rick
    </h2>
    <p>
      unknown
    </p>
    <p>
      Alien
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Amish Cyborg
    </h2>
    <p>
      Dead
    </p>
    <p>
      Alien
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Annie
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Female
    </p>
  </li>
  <li>
    <h2>
      Antenna Morty
    </h2>
    <p>
      Alive
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Antenna Rick
    </h2>
    <p>
      unknown
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
  <li>
    <h2>
      Ants in my Eyes Johnson
    </h2>
    <p>
      unknown
    </p>
    <p>
      Human
    </p>
    <p>
      Male
    </p>
  </li>
</ul>
`;
